python/lsst/ts/externalscripts/auxtel/latiss_take_flats.py

```diff
@@ -0,0 +1,548 @@
+
+        """
+        self.log.debug("Starting Electrometer exposures")
+        lfa_urls = []
+        for i in range(n):
+            await self.electrometer.cmd_startScanDt.set_start(scanDuration=exp_time)
+            lfa_obj = await self.electrometer.evt_largeFileObjectAvailable.next(
+                flush=False, timeout=30
+            )
+            lfa_urls.append(lfa_obj.url)
+
+        return lfa_urls
+
+        # @property
+        # def estimated_atmonochromator_setup_time(self):
+        #     """The estimated time to setup the atmonochromator
+        #     """
+        #     return 50
+
+    async def prepare_summary_table(self):
+        """Prepares final summary table.
+        Checks writing is possible and that s3 bucket can be made
+        """
+
+        # file_output_dir = "/tmp/LatissTakeFlats/"
+        # write folder if required
+        # pathlib.Path().mkdir(parents=True, exist_ok=True)
+        # self.filename =
+        # f"LatissTakeFlats_sequence_summary_{self.group_id}.json"
+
+        # Take a copy as the starting point for the summary
+        self.sequence_summary = {}
+
+        # Add metadata from this script
+        date_begin = utils.astropy_time_from_tai_unix(utils.current_tai()).isot
+        self.sequence_summary["date_begin_tai"] = date_begin
+        self.sequence_summary["script_index"] = self.salinfo.index
+        # Create and empty list for the steps
+        self.sequence_summary["steps"] = []
+
+        # Check to see if in simulations mode, if so provide a mock.
+        if self.bucket is None:
+            self.bucket = salobj.AsyncS3Bucket(
+                salobj.AsyncS3Bucket.make_bucket_name(s3instance=self.s3instance),
+                create=True,
+                domock=self.simulation_mode != 0,
+            )
+
+        # Generate a bucket key
+        self.s3_key_name = self.bucket.make_key(
+            salname="LatissTakeFlats",
+            salindexname=self.salinfo.index,
+            generator="json",
+            date=date_begin,
+            suffix=".json",
+        )
+
+    async def arun(self, checkpoint_active=False, simulation_mode: bool = 0):
+
+        await self.handle_checkpoint(
+            checkpoint_active=checkpoint_active,
+            checkpoint_message="Setting up LATISS",
+        )
+
+        await self._setup_latiss()
+
+        # setup electrometer
+        await self.setup_electrometer()
+
+        # Get sequences for the given setup if not defined explicitly
+        self.log.debug(f"In arun, and {self.sequence=}")
+        if self.sequence is None:
+            self.log.debug("No sequence defined, grabbing default")
+            self.sequence = self.get_flat_sequence(
+                self.config.latiss_filter, self.config.latiss_grating
+            )
+
+        # Prepare summary table for writing
+        await self.prepare_summary_table()
+
+        for i, self.step in enumerate(self.sequence):
```

Comment:
What happens if the script fails in the middle of the sequence? Should the script be run from the beginning again or maybe should it be run with an updated schema (in case a custom sequence is provided) that will take that into account?

If a default sequence is provided, then I guess that the solution would be "run everything again", right? Can somehow the `summary_table` be used to keep track of what was already done, so we can recover the script from that point?